src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingStructureBuilder.java

```diff
@@ -650,8 +650,11 @@ public void checkOrRequestBucket(@Nullable final BuilderBucket requiredResources
             boolean hasOpenRequest = false;
             int count = InventoryUtils.getItemCountInItemHandler(getCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY).orElseGet(null),
               stack -> stack.isItemEqual(itemStack.getItemStack()));
-            int workerInvCount = workerInv ? InventoryUtils.getItemCountInItemHandler(worker.getInventory(), stack -> stack.isItemEqual(itemStack.getItemStack())) : 0;
-            if ((count + workerInvCount) < entry.getValue())
+
+            int totalAmount = neededResources.get(entry.getKey()).getAmount();
```

Comment:
You sure that this can not produce an NPE?

Reply:
Should not be possible to happen, but I'll build a safeguard in